source/extensions/filters/network/dubbo_proxy/router/route_matcher.cc

```diff
@@ -179,16 +195,62 @@ SingleRouteMatcherImpl::SingleRouteMatcherImpl(const RouteConfig& config,
   ENVOY_LOG(debug, "dubbo route matcher: routes list size {}", routes_.size());
 }
 
+bool SingleRouteMatcherImpl::matchServiceGroup(RpcInvocationImpl& invocation) const {
+  if (!group_.has_value() || group_.value().empty()) {
+    return true;
+  }
+
+  // Since the configured group is not empty, we need to parse the attachment of the invocation and
+  // get the group in it.
+  invocation.mutableAttachment();
```

Comment:
Can we decouple setting the service group from getting a mutable attachment (i.e. setting it lazily in `RpcInvocationImpl::serviceGroup()`)? If that's possible then we could mark `mutableAttachment()` as `const` (and `RpcInvocationImpl::attachments_` as `mutable` since it's just a cached value) and get rid of the const casts above.

Reply:
I think I can have a try.